src/Compiler/TypedTree/TcGlobals.fs

```diff
@@ -896,6 +900,18 @@ type TcGlobals(
           | [||] -> mkILCustomAttrsFromArray generatedAttributes
           | attrs -> mkILCustomAttrsFromArray (Array.append attrs generatedAttributes)
 
+  let addValGeneratedAttrs (v: Val) m =
+      if not noDebugAttributes then
+          let attrs = [
+              Attrib(v_CompilerGeneratedAttribute_tcr, ILAttrib compilerGeneratedAttribute.Method.MethodRef, [], [], false, None, m)
+              Attrib(v_DebuggerNonUserCodeAttribute_tcr, ILAttrib debuggerNonUserCodeAttribute.Method.MethodRef, [], [], false, None, m)
+              Attrib(v_DebuggerNonUserCodeAttribute_tcr, ILAttrib debuggerNonUserCodeAttribute.Method.MethodRef, [], [], true, None, m)
+          ]
```

Comment:
Wish this list could be cached and not recreated all the time, but one of the parameters is range (of the union case), though I am not quite sure what purpose it serves.